File: Submissions/#FF_div2/A/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    int p = sc.nextInt();
    int n = sc.nextInt();
    
    int answer = -1;

    int[] h = new int[p];

    Arrays.fill(h, -1);

    for (int i = 0; i < n; i++) {
      int x = sc.nextInt();
      if (h[x % p] == x % p) {
        if (answer == -1) {
          answer = i + 1;
        }
      } else {
        h[x % p] = x % p;
      }
    }
    sc.close();

    System.out.println(answer);

  }

}

File: Submissions/#FF_div2/A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  int p, n;
  cin >> p >> n;

  int answer = -1;

  vector<int> h(p);

  fill(h.begin(), h.end(), -1);

  for (int i = 0; i < n; i++) {
    int x;
    cin >> x;
    if (h[x % p] == x % p) {
      if (answer == -1) {
        answer = i + 1;
      }
    } else {
      h[x % p] = x % p;
    }
  }

  cout << answer << endl;

}